app/rankers/algorithms.py

```diff
@@ -83,25 +97,34 @@ def thompson_sampling(
         return recs
 
     alpha_prior, beta_prior = DEFAULT_ALPHA_PRIOR, DEFAULT_BETA_PRIOR
-    if clk_data and 'default' in clk_data:
+    if metrics and 'default' in metrics:
         # 'default' is a special key we can use for anything that is missing.
         # The values here aren't actually clicks or impressions,
         # but instead direct alpha and beta parameters for the module CTR prior
-        alpha_prior, beta_prior = clk_data['default'].clicks, clk_data['default'].impressions
+        alpha_prior = metrics['default'].trailing_7_day_opens
+        beta_prior = metrics['default'].trailing_7_day_impressions
         if alpha_prior < 0 or beta_prior < 0:
-            logging.error("Alpha (%s) or Beta (%s) prior < 0 for module %s", alpha_prior, beta_prior, clk_data['default'].mod_item)
+            logging.error(
+                f"Alpha {alpha_prior} or Beta {beta_prior} prior < 0 for module {metrics['default'].id}")
             alpha_prior, beta_prior = DEFAULT_ALPHA_PRIOR, DEFAULT_BETA_PRIOR
 
     scores = []
     # create prior distribution for CTR from parameters in click data table
     prior = beta(alpha_prior, beta_prior)
     for rec in recs:
-        resolved_id = rec.item.item_id
-        d = clk_data.get(resolved_id)
+        try:
+            # Recommendation is keyed on item_id.
+            clickdata_id = rec.item.item_id
+        except AttributeError:
+            # Slates are keyed on their id.
+            clickdata_id = rec.id
```

Comment:
Why are we catching the exception here?

Reply:
Answer: This is how slates are distinguished from recommendations.

Context: The engagement data pipeline identifies recommendations based on item id, and slates based on slate id.

Maybe directly checking against the type would be cleaner.